modalities/dom/components/elements/magenta-visualizer.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import Xen from '../xen/xen.js';
+
+class MagentaVisualizer extends Xen.Base {
+  get template() {
+    return Xen.html`
+      <canvas id="canvas" width="400" height="300"></canvas>
+      <button id="start" on-click="onStart"> Start </button>
+      <button id="stop" on-click="onStop"> Stop </button>
+    `;
+  }
+
+  static get observedAttributes() {
+    return ['notes'];
+  }
+
+  _didMount() {
+    this.canvas = this.host.getElementById('canvas');
+  }
+
+  _render(props, state) {
+    this.initVisualizer(props);
+    this.initPlayer();
+    return {...state};
+  }
+
+  initPlayer() {
+    if (this.visualizer && !this.player) {
+      this.player = new window.mm.Player(true, {
+        run: (note) => this.visualizer.redraw(note),
+        stop: () => {console.log('done');}
+      });
+    }
+  }
+
+  onStart() {
+    this.player.start(this.notes);
+  }
+
+  onStop() {
+    this.player.stop();
+  }
+
+  initVisualizer(props) {
+    if (props.notes && !this.notesSet) {
+      this.visualizer = new window.mm.Visualizer(this.notes, this.canvas);
+      this.notes = props.notes;
+      this.notesSet = true;
```

Comment:
rename: notesSet --> isNotesSet

Reply:
We can also investigate why this.notes was truthy in the first place, so we don't have to create a new variable (we ran out of time yesterday). 
We can time box this investigation.